ViTool/Models/MirrorAlgorithm.cs

```diff
@@ -0,0 +1,222 @@
+                IsRunning = false;
+                return null;
+            }
+
+            mirroredImgDirectory = Path.Combine(Path.GetFullPath(Path.Combine(directory, @"..\")), Path.GetFileName(directory) + "Mirrored");
+
+            if (Directory.Exists(mirroredImgDirectory))
+                Directory.Delete(mirroredImgDirectory, true);
+
+            Directory.CreateDirectory(mirroredImgDirectory);
+
+            return mirroredImgDirectory;
+        }
+
+        async Task ProcessFilesParalelAsync(string[] files, string mirroredImgDirectory, IProgress<ProgressReportModel> progress)
+        {
+            int procesorsCount = Environment.ProcessorCount;
+
+            if (procesorsCount >= 4)
+                procesorsCount -= 2;
+
+            await Task.Run(() => Parallel.ForEach<string>(files, new ParallelOptions { MaxDegreeOfParallelism = procesorsCount }, src =>
+              {
+
+                  if (Path.GetExtension(src) == xmlExt)
+                      SaveXml(src, mirroredImgDirectory, xmlExt, FlipXml(src));
+
+                  if (Path.GetExtension(src) == imgExt)
+                  {
+                      Stopwatch watch = new Stopwatch();
+                      watch.Start();
+                      SaveImg(src, mirroredImgDirectory, imgExt, FlipImg(src));
+
+                      Output.Add(src);
+
+                      watch.Stop();
+
+                      SendProgressReport(progress, src, watch);
+                  }
+
+              }));
+        }
+
+        private void SendProgressReport(IProgress<ProgressReportModel> progress, string src, Stopwatch watch)
+        {
+            ProgressReportModel progressReportModel = new ProgressReportModel();
+
+            progressReportModel.FilesProcessed.Add(src);
+            progressReportModel.NumberOfAllFilesToProcess = HowMuchThereIs;
+            progressReportModel.PercentageComplete = (Output.Count * 100) / HowMuchThereIs;
+            progressReportModel.TimeConsumedByProcessedFiles = watch.Elapsed.TotalMilliseconds * 0.001;
+
+            progress.Report(progressReportModel);
+        }
+
+        XmlDocument FlipXml(string imgSrc)
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.Load(imgSrc);
+            int frameWidth = int.Parse(doc.DocumentElement.SelectSingleNode("/annotation/size/width").InnerText);
+
+            var nodes = doc.DocumentElement.SelectNodes("/annotation/object");
+
+            foreach (XmlNode currentNode in nodes)
+            {
+                int newXmin = frameWidth - int.Parse(currentNode.SelectSingleNode("bndbox/xmin").InnerText);
+                XmlNode node = currentNode.SelectSingleNode("bndbox/xmin");
+                node.InnerText = newXmin.ToString();
+
+                int test = int.Parse(currentNode.SelectSingleNode("bndbox/xmax").InnerText);
+                int newXmax = frameWidth - test;
+                XmlNode xmax = currentNode.SelectSingleNode("bndbox/xmax");
```

Comment:
you are using those bndbox/xmax and xmin a lot, consider adding enum for them, if you change for some reason name of those nodes you would change those names only in one place